test/unit/foreman/renderer/template_snapshot_service.rb

```diff
@@ -0,0 +1,31 @@
+class TemplateSnapshotService
+  SNAPSHOTS_DIRECTORY = Rails.root.join('test', 'unit', 'foreman', 'renderer', 'snapshots')
+
+  def self.templates
+    new.templates
+  end
+
+  def self.host
+    new.host
+  end
+
+  def templates
+    template_names = ProvisioningTemplatesList.seeded_templates.select {|item| item[:snapshot_test] }.pluck(:name)
+    template_names.map { |template_name| Template.find_by!(name: template_name) }
+  end
+
+  def host
+    host = Host::Managed.new(name: 'snapshotshost')
+    host.architecture = Architecture.find_by!(name: 'x86_64')
+    host.operatingsystem = Operatingsystem.find_by!(name: 'Redhat')
+    host.subnet = Subnet.find_by!(name: 'one')
+    host.medium = Medium.find_by!(name: 'CentOS 5.4')
```

Comment:
I'd prefer CentOS 7.0 here if we have a fixture for that, latest stable version of Red Hat is the one we should be watching for. Ideally all OSes but that's just too slow to test.

Reply:
there is only CentOS 5.4 in the fixtures

Reply:
We are in process of getting rid of fixtures, this is 3 years process or even more, we slowly refactor to FactoryBot. Is there any problem with simply not using fixtures for snapshoting host stub? I think it should be easy simply to use FactoryBot for everything (except perhaps Settings that's acceptable).